### PR TITLE
Bug fix : missing C# Swig support for ETP

### PR DESCRIPTION
**Pull request by mpoudret**
What does this implement/fix? Explain your changes.
---------------------------------------------------
- Bug fix : missing C# Swig support for ETP
-  adds C# EtpClientExample

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, etc?
-------------------------------------
No

Any other comments?
-------------------
No

Where has this been tested?
---------------------------
**Operating System:** Windows 10 64bits

**Platform:** Visual Studio Community 2019

### CODE REVIEW

File: cmake/MyOwnClientCoreHandlers.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	{
+	}
+
+	public override void on_OpenSession(OpenSession os, long correlationId)
+	{
+		Console.WriteLine("Opened a session");
+
+		Console.WriteLine("Asking all dataobjects");
+		GetResources mb = new GetResources();
+		mb.context.uri ="eml:///";
```
Comment (philippeVerney) [inline cmake/MyOwnClientCoreHandlers.cs:20]: missing space after equal

File: cmake/MyOwnStoreProtocolHandlers.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            Console.WriteLine("datatype : " + dataObj.resource.dataObjectType);
+            Console.WriteLine("name : " + dataObj.resource.name);
+            Console.WriteLine("xml : " + dataObj.data);
+            Console.WriteLine("*************************************************");
+        }
+
+        Console.WriteLine("Asking to close");
+        getSession().close();
+    }
+}
```
Comment (philippeVerney) [inline cmake/MyOwnStoreProtocolHandlers.cs:30]: missing new line at and of file

File: cs/CMakeLists.txt [hunk 3]
```diff
@@ -30,17 +36,9 @@ endif (WITH_ETP)
 
 # SWIG execution
 if (SWIG_LINKED_TO_RELEASE)
-	if (WITH_RESQML2_2)
-		set ( EXECUTE_COMMAND "${SWIG_EXECUTABLE} -DWITH_RESQML2_2 -c++ -csharp -dllimport ${CPP_LIBRARY_NAME}${CMAKE_RELEASE_POSTFIX}.${Fesapi_VERSION}.dll -namespace f2i.energisticsStandardsApi -o ${FESAPI_ROOT_DIR}/swig/swigGeneratedCsWrapper.cpp -outdir ${FESAPI_ROOT_DIR}/cs/src ${FESAPI_ROOT_DIR}/swig/swigModule.i")
-	else (WITH_RESQML2_2)
-		set ( EXECUTE_COMMAND "${SWIG_EXECUTABLE} -c++ -csharp -dllimport ${CPP_LIBRARY_NAME}${CMAKE_RELEASE_POSTFIX}.${Fesapi_VERSION}.dll -namespace f2i.energisticsStandardsApi -o ${FESAPI_ROOT_DIR}/swig/swigGeneratedCsWrapper.cpp -outdir ${FESAPI_ROOT_DIR}/cs/src ${FESAPI_ROOT_DIR}/swig/swigModule.i")
-	endif(WITH_RESQML2_2)
+	set ( EXECUTE_COMMAND "${SWIG_EXECUTABLE} ${SWIG_DEFINITION} -c++ -csharp -dllimport ${CPP_LIBRARY_NAME}${CMAKE_RELEASE_POSTFIX}.${Fesapi_VERSION}.dll -namespace f2i.energisticsStandardsApi -o ${FESAPI_ROOT_DIR}/swig/swigGeneratedCsWrapper.cpp -outdir ${FESAPI_ROOT_DIR}/cs/src ${FESAPI_ROOT_DIR}/swig/swigModule.i")
```
Comment (philippeVerney) [inline cs/CMakeLists.txt:39]: Why to delete this condition on WITH_RESQML2_2? Is it no more necessary?

File: cs/EtpClientExample/EtpClientExample.csproj [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    <UpdatePeriodically>false</UpdatePeriodically>
+    <UpdateRequired>false</UpdateRequired>
+    <MapFileExtensions>true</MapFileExtensions>
+    <ApplicationRevision>0</ApplicationRevision>
+    <ApplicationVersion>1.0.0.%2a</ApplicationVersion>
+    <IsWebBootstrapper>false</IsWebBootstrapper>
+    <UseApplicationTrust>false</UseApplicationTrust>
+    <BootstrapperEnabled>true</BootstrapperEnabled>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
```
Comment (philippeVerney) [inline cs/EtpClientExample/EtpClientExample.csproj:31]: We do not support 32bits configuration.

File: cs/EtpClientExample/EtpClientExample.sln [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.27130.2024
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "EtpClientExample", "EtpClientExample.csproj", "{45AAFC55-D20F-46C6-924B-5D97CFB80CCA}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "fesapiCs", "..\fesapiCs.csproj", "{50CF64AB-EA6E-4D04-AC83-6DFEC148A4F0}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
```
Comment (philippeVerney) [inline cs/EtpClientExample/EtpClientExample.sln:11]: We do not support 32bits configuration.

File: cs/EtpClientExample/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// Les informations générales relatives à un assembly dépendent de 
+// l'ensemble d'attributs suivant. Changez les valeurs de ces attributs pour modifier les informations
+// associées à un assembly.
+[assembly: AssemblyTitle("example")]
```
Comment (philippeVerney) [inline cs/EtpClientExample/Properties/AssemblyInfo.cs:8]: EtpClientExample would have bee better I think.

File: cs/EtpClientExample/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// Les informations générales relatives à un assembly dépendent de
```
Comment (philippeVerney) [inline cs/EtpClientExample/Properties/AssemblyInfo.cs:5]: In english would have been better.

**Review by philippeVerney — CHANGES_REQUESTED** [review]
Thanks for the PR.
Only "Why to delete this condition on WITH_RESQML2_2? Is it no more necessary?" may request a change.
Other comments are much more detail especially regarding the fact that some of these comments also apply to already integrated similar files (other csproj for example)

Last comment, adding an unit test would be awesome though not required.

File: cmake/MyOwnClientCoreHandlers.cs (hunk 1 above)
Comment (mpoudret) [reply cmake/MyOwnClientCoreHandlers.cs:20]: Done.

File: cmake/MyOwnStoreProtocolHandlers.cs (hunk 2 above)
Comment (mpoudret) [reply cmake/MyOwnStoreProtocolHandlers.cs:30]: Done.

File: cs/EtpClientExample/Properties/AssemblyInfo.cs (hunk 6 above)
Comment (mpoudret) [reply cs/EtpClientExample/Properties/AssemblyInfo.cs:8]: Done.

File: cs/EtpClientExample/Properties/AssemblyInfo.cs (hunk 7 above)
Comment (mpoudret) [reply cs/EtpClientExample/Properties/AssemblyInfo.cs:5]: Done.

File: cs/CMakeLists.txt (hunk 3 above)
Comment (mpoudret) [reply cs/CMakeLists.txt:39]: This condition is included in lines 30 to 32 thanks to the SWIG_DEFINITION variable.

File: cs/EtpClientExample/EtpClientExample.csproj (hunk 4 above)
Comment (mpoudret) [reply cs/EtpClientExample/EtpClientExample.csproj:31]: Done. As well as the C# example.

File: cs/EtpClientExample/EtpClientExample.sln (hunk 5 above)
Comment (mpoudret) [reply cs/EtpClientExample/EtpClientExample.sln:11]: Done. As well as the C# example.

**Review by philippeVerney — APPROVED** [review]

**Pull request merged**